### PR TITLE
fix(sdk.v2): Fix the bug where pipeline_parameters is not used.

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
Fixes:
- The `Compiler().compile` method accepts `pipeline_parameters` as an argument but doesn't actually use it.
- The compiler CLI tool does not even expose an interface to accept pipeline parameters.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [x] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: sdk/python/kfp/v2/compiler/main.py [hunk 1]
```diff
@@ -102,14 +106,20 @@ def __exit__(self, *args):
     dsl._pipeline._pipeline_decorator_handler = self.old_handler
 
 
-def compile_pyfile(pyfile, function_name, pipeline_root, output_path, type_check):
+def compile_pyfile(pyfile, function_name, pipeline_root, pipeline_parameters,
```
Comment (numerology) [inline sdk/python/kfp/v2/compiler/main.py:109]: Consider type-hinting this function?

File: sdk/python/kfp/v2/compiler/main.py [hunk 2]
```diff
@@ -52,7 +55,7 @@ def parse_arguments():
 
 
 def _compile_pipeline_function(pipeline_funcs, function_name, pipeline_root,
-                               output_path, type_check):
+                               pipeline_parameters, output_path, type_check):
```
Comment (numerology) [inline sdk/python/kfp/v2/compiler/main.py:58]: Consider type-hinting this function as well?

**Review by numerology — COMMENTED** [review]
Thanks Chen! Overall LGTM

File: sdk/python/kfp/v2/compiler/main.py (hunk 1 above)
Comment (chensun) [reply sdk/python/kfp/v2/compiler/main.py]: Done.

File: sdk/python/kfp/v2/compiler/main.py (hunk 2 above)
Comment (chensun) [reply sdk/python/kfp/v2/compiler/main.py]: Done.

**Pull request merged**